File: javascript/src/museo/visitante.js
export class Visitante {
  /**
   * @param {string} tipoConsulta
   * @param {import('./monitorVestibulo.js').MonitorVestibulo} monitor
   */
  constructor(tipoConsulta, monitor) {
    this.tipoConsulta = tipoConsulta;
    this.monitor = monitor;
  }

  getTipoConsulta() {
    return this.tipoConsulta;
  }

  getMonitor() {
    return this.monitor;
  }

  consultarObrasPorSala(idSala) {
    return this.monitor.mostrarListadoPorSala(idSala);
  }
}

File: java/src/museo/Visitante.java
package museo;

import java.util.List;

public class Visitante {

    private final String tipoConsulta;
    private final MonitorVestibulo monitor;

    public Visitante(String tipoConsulta, MonitorVestibulo monitor) {
        this.tipoConsulta = tipoConsulta;
        this.monitor = monitor;
    }

    public String getTipoConsulta() {
        return tipoConsulta;
    }

    public MonitorVestibulo getMonitor() {
        return monitor;
    }

    public List<ObraArte> consultarObrasPorSala(int idSala) {
        return monitor.mostrarListadoPorSala(idSala);
    }
}
